File: tests/fixtures/python/StreamFlatMap.py
from __future__ import annotations


class StreamFlatMap:
    class Item:

        def get_tags(self) -> list[str]:
            return None

    def flatten_nested(self, nested: list[list[str]]) -> list[str]:
        return [nested_item for nested in nested for nested_item in nested]

    def collect_tags(self, items: list[Item]) -> set[str]:
        return {item_item for item in items for item_item in item.get_tags()}

File: tests/fixtures/java/StreamFlatMap.java
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public class StreamFlatMap {
    static class Item {
        List<String> getTags() {
            return null;
        }
    }

    public List<String> flattenNested(List<List<String>> nested) {
        return nested.stream()
                .flatMap(group -> group.stream())
                .collect(Collectors.toList());
    }

    public Set<String> collectTags(List<Item> items) {
        return items.stream()
                .flatMap(item -> item.getTags().stream())
                .collect(Collectors.toSet());
    }
}
